File: passwordKalculator_Java/Password/Password.java
package Password;
import StartingData.StartingData;
import java.util.Random;
public abstract class Password{
	private String pwd;
	private int len;
	public Password(String p, int l){
		pwd=p;
		len=l;
	}
	public String getPassword(){
		return pwd;
	}
	public void setPassword(String s){
		pwd=s;
	}
	public int getLength(){
		return len;
	}
	public void setLength(int l){
		len=l;
	}
	public String inversion(String s){
		char v[] =s.toCharArray();
		int n=v.length;
		for(int i=0;i<n/2;++i){
			char tmp=v[i];
			v[i]=v[n-i-1];
			v[n-i-1]=tmp;
		}
		return new String(v);
	}

	public abstract String combination1(StartingData d1, StartingData d2);
    public abstract String combination2(StartingData d1, StartingData d2);
    public abstract String combination3(StartingData d1, StartingData d2);
    
	public abstract String getDescription();
	//getDescription mi serve virtuale perchè con l'oggetto result che ha TS Password(madre)
    //devo chiamare i giusti getPassword in base al TD istanziato
}

File: passwordKalkulator_C++/password.cpp
#include "password.h"
#include<iostream>
using namespace std;
Password::Password(QString p, int l) : pwd(p), len(l) {}

Password::~Password() {}

QString Password::getPassword() const{
    return pwd;
}
void Password::setPassword(QString p){
    pwd=p;
}
int Password::getLength() const{
    return len;
}
void Password::setLength(int l){
    len=l;
}

QString Password::inversion(QString s){
    string str=s.toStdString();
    int n=str.length();
    for(int i=0; i<n/2;++i){
        char tmp=str[i];
        str[i]=str[n-i-1];
        str[n-i-1]=tmp;
    }
    return QString::fromStdString(str);

}
